### PR TITLE
 fine-tuning implementation

### PR DESCRIPTION
**Pull request by zachfox**
### Summary:
- **Adapter Layers Generation**: Modified the code to use the pretrained configuration file to generate the fine-tuning configuration for adapter layers. 
- **Pretrained Config during Fine-Tuning**: Updated the fine-tuning process to use the pretrained configuration file to correctly load the pretrained model along with the adapter layers.

### Changes:
1. `yaml_to_config.py`: 
    - Added logic to initialize the fine-tuning configuration using the pretrained config.
2. `fine_tune.py`: 
    - Modified the model loading function to ensure the pretrained model is loaded with the fine-tuning 
    - Updated the fine-tuning procedure to use the correct configuration.

### CODE REVIEW

**Comment by zachfox** [comment]
This method has successfully loaded for `clintox` and `tdc/cyp_cyp3a4_veith`.

**Comment by zachfox** [comment]
### Future Enhancements/To-Do:
- [ ] **Classifier Loss Function Integration**  
    - Integrate a loss function specific to classification tasks, ensuring it works with the current fine-tuning framework and pretrained models. This will require simple changes to `hydragnn`. 
  
- [ ] **Multiclass Parsing**  
    - Implement logic to handle multiclass data during both training and evaluation phases. 
    - Not sure how to do this with the currrent `.yaml` files - may need to add metadata corresponding to number of classes into the `.yaml` descriptions.  
    - Ensure the model can output correct probabilities and predictions for multiple classes.

- [ ] **Pretraining Directory Support**  
    - Add the ability to point to a directory containing both pretrained model weights and configuration files.
    - Update model loading logic to check this directory and use both files/integrate with fine-tuning.

**Comment by allaffa** [comment]
@zachfox 
Is this something ready to be reviewed or is it still a WIP?
